src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java

```diff
@@ -32,98 +28,111 @@
-                .column("my_column", type).build();
-        this.createdTables.add(table);
-        refreshVirtualSchema();
-        final String tableName = virtualSchema.getFullyQualifiedName() + "." + POSTGRES_TABLE_NAME;
-        return new VirtualSchemaTestTable<>(tableName, getTruncateValueSetter(table));
+    @Override
+    protected SingleValueVirtualSchemaTestTable<Boolean> createBooleanVirtualSchemaTable() {
+        return createTestTable("boolean");
     }
 
-    private <T> VirtualSchemaTestTable.SingleRowTableProvisioner<T> getTruncateValueSetter(final Table table) {
-        return (value) -> {
-            truncateTable(table);
-            table.insert(value);
+    private <T> SingleValueVirtualSchemaTestTable<T> createTestTable(final String type) {
+        return new SingleValuePostgresVirtualSchemaTestTable<>() {
+            @Override
+            protected Table createTable(final Schema schema) {
+                return schema.createTableBuilder(getUniqueIdentifier())//
+                        .column("my_column", type).build();
+            }
         };
     }
 
-    @Override
-    protected String createBooleanVirtualSchemaTableWithSingleTrueValue() throws SQLException {
-        final Table table = postgresSchema.createTableBuilder(POSTGRES_TABLE_NAME)//
-                .column("boolean", "boolean").build()//
-                .insert(true);
-        this.createdTables.add(table);
-        refreshVirtualSchema();
-        return virtualSchema.getFullyQualifiedName() + "." + POSTGRES_TABLE_NAME;
+    protected Connection createExasolConnection() throws SQLException {
+        return SETUP.getExasolContainer().createConnection();
     }
 
     @Override
-    protected VirtualSchemaTestTable<String> createStringVirtualSchemaTable() throws SQLException {
+    protected SingleValueVirtualSchemaTestTable<String> createStringVirtualSchemaTable() {
         return createTestTable("VARCHAR(500)");
     }
 
-    private void truncateTable(final Table table) throws SQLException {
-        SETUP.getPostgresqlStatement().executeUpdate("TRUNCATE TABLE " + table.getFullyQualifiedName());
-    }
-
-    @Override
-    protected void deleteCreatedTables() {
-        this.createdTables.forEach(Table::drop);
-        this.createdTables.clear();
+    private static abstract class PostgresVirtualSchemaTestTable implements VirtualSchemaTestTable {
+        private final VirtualSchema virtualSchema;
+        private final Table table;
+        private final Schema postgresqlSchema;
+
+        public PostgresVirtualSchemaTestTable() {
+            this.postgresqlSchema = SETUP.getPostgresFactory().createSchema(getUniqueIdentifier());
+            this.table = createTable(this.postgresqlSchema);
+            this.virtualSchema = SETUP.createVirtualSchema(this.postgresqlSchema.getName(), Map.of());
+        }
+
+        protected abstract Table createTable(Schema schema);
```

Comment:
This is dangerous as there's an implicit restriction (that the user is not aware of) that the Schema passed has to be the same to the Schema created in the constructor, but at the same time this method can be called with a completely different Schema which will lead to unexpected behaviors.
I think we need a different approach here.

Reply:
the method is not called from outside of the construction. Since it's protected it is no public API.
It's just delegating part of the Implementation to the implementing class.

Reply:
it's not true. It still can be call with a different schema (even from the same class). It should be not accepting a parameter and use the instance variable instead.